addons/udes_stock/controllers/stock_picking_batch.py

```diff
@@ -201,7 +203,20 @@ def reserve_pallet(self, ident, pallet_name):
             raise ValidationError(
                 _("The specified batch is not assigned to you."))
 
-        batch.reserve_pallet(pallet_name)
+        # TODO this is messy
+        if picking_id is not None:
+            picking = None
```

Comment:
You can just get this out of the if and inside the if: `picking = StockPicking.browse(int(picking_id))`